go/vt/schemamanager/schemaswap/schema_swap.go

```diff
@@ -75,6 +90,290 @@ type shardSchemaSwap struct {
 	healthWaitingChannel *chan interface{}
 }
 
+// shardSwapMetadata contains full metadata about schema swaps on a certain shard.
+type shardSwapMetadata struct {
+	shardName string
+	err       error
+	// IDs of last started and last finished schema swap on the shard
+	lastStartedSwap, lastFinishedSwap uint64
+	// currentSQL is the SQL of the currentlr running schema swap (if there is any)
+	currentSQL string
+}
+
+// Run is the main entry point of the schema swap process. It drives the process from start
+// to finish, including possible restart of already started process. In the latter case the
+// method should be just called again and it will pick up already started process. The only
+// input argument is the SQL statements that comprise the schema change that needs to be
+// pushed using the schema swap process.
+func (schemaSwap *Swap) Run(sql string) error {
+	err := schemaSwap.createShardObjects()
+	if err != nil {
+		return err
+	}
+	err = schemaSwap.initializeSwap(sql)
+	if err != nil {
+		return err
+	}
+	err = schemaSwap.runOnAllShards(
```

Comment:
How about using a different variable name than err, here? Like:
errHealth := ....
Then the if err != nil below would be a lot more readable.